File: WeddingInfo.Frontend/src/app/models/Location.ts
import { ILocation } from './interfaces/ILocation';
import { IMedia } from './interfaces/IMedia';

export class Location implements ILocation {
    public address: string;
    public description: string;
    public id: number;
    public images: IMedia[];
    public name: string;
    public order: number;
    public videos: IMedia[];
    public website: string;

    public constructor(location?: ILocation) {
        this.address = location && location.address || '';
        this.description = location && location.description || '';
        this.id = location && location.id || 0;
        this.images = location && location.images || [];
        this.name = location && location.name || '';
        this.order = location && location.order || 0;
        this.videos = location && location.videos || [];
        this.website = location && location.website || '';
    }
}

File: WeddingInfo.Domain/Models/Location.cs
using System;
using System.Collections.Generic;

namespace WeddingInfo.Domain.Models
{
    public class Location
    {
		public int Id { get; set; }
		public string Name { get; set; }
		public string Address { get; set; }
        public int Order { get; set; }
        public string Website { get; set; }
		public string Description { get; set; }
		public IList<Media> Images { get; set; } = new List<Media>();
		public IList<Media> Videos { get; set; } = new List<Media>();
    }
}
